specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,266 @@
+throw an exception.
+
+# API Pages
+
+## UserActivityHistory class
+
+This class provides static methods that enable you to query the user's activity history.
+This activity history is stored in a database managed by a local service, and these APIs
+call out to that service to retrieve data from the database.
+
+Here is an example usage of the class that will enable you to bring back the webpage for a
+Korean recipe that the user had previously interacted with within the last day:
+
+```c#
+UserActivityHistoryQuery query = new();
+query.Keywords = new string[] { "Korean", "recipe" };
+query.LatestStartTime = DateTime.Now.AddDays(-1);
+
+IList<UserActivityHistoryItem> results = UserActivityHistory.Search(
+    new UserActivityHistoryQuery[] { query },
+    UserActivityHistoryOrderBy.DwellTime,
+    maxResults: 1);
+
+UserActivityHistoryItem item = results.FirstOrDefault();
+
+if (item != null)
+{
+    // Now we can use item.ActivationUri to bring back the webpage in the state in which the user
+    // was last viewing it.
+}
+```
+
+## UserActivityHistory.Search method
+
+This method synchronously queries the user's activity history and returns a list of items matching
+the criteria specified in the `queries` parameter. The results are ordered in descending order
+according by the `orderBy` parameter: either by the most recent start times, the most recent
+end times, or the longest time spent on the activity.
+
+The results will include the user activity history items that match any one of the queries passed in.
+To match a given query, the item must match all of the criteria specified in that query.  Any query
+property that is left empty or null will be ignored.  A case-insensitive lexical search will be
+performed on the keywords in the query, which will match if all of the keywords are found somewhere
+in the DisplayText property of the item.
+
+This method will not perform any parsing of the keywords for semantic meaning or natural language -
+it is expected that the app will have already performed that step and will pass the result of that
+into this API.
+
+A case-insensitive lexical search will also be performed on the ContentType property of the item,
+which is the [MIME type](https://docs.w3cub.com/http/basics_of_http/mime_types/complete_list_of_mime_types.html)
+of the resource the user was interacting with.  The search will match if the ContentType of the item matches
+the ContentType property of the query.  The ContentType supports using an asterisk as a wildcard
+to match a range of content types - e.g., "image/*" will match "image/png", "image/jpeg", etc.
+
+## UserActivityHistory.GetAppsWithUserActivity method
+
+This method synchronously retrieves a list of all the app names with data in the user's activity
+history database. You can use this, for example, to show the user the list of apps that are being
+queried against, so the user can understand why an app that is not recording user activity is not
```

Comment:
>...why an app not recording isn't showing up in the list of apps that recorded...

Typo? Should be "...what an app <strike>that is not</strike> recording user activity is not showing up..." ?